airflow/providers/http/hooks/http.py

```diff
@@ -385,8 +387,9 @@ async def run(
             for attempt in range(1, 1 + self.retry_limit):
                 response = await request_func(
                     url,
-                    json=data if self.method in ("POST", "PUT", "PATCH") else None,
                     params=data if self.method == "GET" else None,
+                    data=data if self.method in ("POST", "PUT", "PATCH") else None,
```

Comment:
```suggestion
                    data=data,
```
do we really need that logic?  shouldn't we do as the user asks?